tests/by-util/test_touch.rs

```diff
@@ -1013,3 +1013,26 @@ fn test_touch_f_option() {
     assert!(at.file_exists(file));
     at.remove(file);
 }
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_touch_non_utf8_paths() {
+    use std::ffi::OsStr;
+    use std::os::unix::ffi::OsStrExt;
+
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    // Create a test file with non-UTF-8 bytes in the name
+    let non_utf8_bytes = b"test_\xFF\xFE.txt";
+    let non_utf8_name = OsStr::from_bytes(non_utf8_bytes);
+
+    // Test that touch handles non-UTF-8 file names without crashing
+    let result = scene.ucmd().arg(non_utf8_name).succeeds();
+
+    // Verify no output and file was created
+    result.no_output();
```

Comment:
The `result` variable is not really necessary:
```suggestion
    scene.ucmd().arg(non_utf8_name).succeeds().no_output();
```